test/e2e/operator.go

```diff
@@ -293,13 +293,12 @@ var _ = XDescribe("ARO Operator - MachineSet Controller", func() {
 		Expect(err).NotTo(HaveOccurred())
 		Expect(mss.Items).NotTo(BeEmpty())
 
-		// Zero all machinesets (avoid availability zone confusion)
+		// Zero all machinesets, wait a few seconds for operator
 		for _, object := range mss.Items {
 			err = scale(object.Name, 0)
 			Expect(err).NotTo(HaveOccurred())
 
-			err = waitForScale(object.Name)
-			Expect(err).NotTo(HaveOccurred())
+			time.Sleep(5 * time.Second)
```

Comment:
Sleep in tests is usually a sign of a problem. Why we need sleep here?

Reply:
> If we use just this, the operator isn't able to reconcile the replicas quickly enough, and the test fails at `Expect(replicaCount).To(BeEquivalentTo(minSupportedReplicas))`

So we are bascially giving the operator 5 seconds to reconcile? Sounds super unreliable. There is no guarantee that operator is going to reconcile in specific amount of time: there are many tihngs in play here (rate limiting, for example). It might be the case it takes longer than 5 seconds.

This has a big potential for another flake.


> So sleep is meant to give the operator time to reconcile in a way that doesn't take more than a little extra time. The waitForScale was making this a 20 minute test 😄

Can we fire scale in the loop and have `waitForScale` in a separate loop? Similar to what we do below? Will it speed things up? It will not be 5 seconds, but it will be reliable which is more important in this case, IMO.

```go
		for _, ms := range mss.Items {
			err = waitForScale(ms.Name)
			Expect(err).NotTo(HaveOccurred())
		}
```